README.md

```diff
@@ -1,103 +1,122 @@
+* Value type is [string](https://www.elastic.co/guide/en/logstash/current/configuration-file-structure.html#string)
+* Required
 
-- Build your plugin gem
+This is the [Dynatrace API token](https://www.dynatrace.com/support/help/dynatrace-api/basics/dynatrace-api-authentication/) which will be used to authenticate log ingest requests.
+It requires the `logs.ingest` (Ingest Logs) scope to be set and it is recommended to limit scope to only this one.
+Example: `"api_key" => "dt0c01.4XLO3..."`
 
-```sh
-gem build logstash-output-dynatrace.gemspec
+### `ssl_verify_none`
 
-```
+* Value type is [boolean](https://www.elastic.co/guide/en/logstash/current/configuration-file-structure.html#boolean)
+* Optional
+* Default value is `false`
 
-- Install the plugin from the Logstash home
+It is recommended to leave this optional configuration set to `false` unless absolutely required.
+Setting `ssl_verify_none` to `true` causes the output plugin to skip certificate verification when sending log ingest requests to SSL and TLS protected HTTPS endpoints.
+This option may be required if you are using a self-signed certificate, an expired certificate, or a certificate which was generated for a different domain than the one in use.
 
-```sh
-# Logstash 2.3 and higher
-bin/logstash-plugin install --no-verify
+### `codec` 
 
-# Prior to Logstash 2.3
-bin/plugin install --no-verify
-```
+* Value type is codec
+* Default value is "plain"
+* 
+The codec used for output data. Output codecs are a convenient method for encoding your data before it leaves the output without needing a separate filter in your Logstash pipeline.
 
-- Start Logstash and proceed to test the plugin
+### `enable_metric`
 
-## Contributing
+* Value type is boolean
+* Default value is true
 
-All contributions are welcome: ideas, patches, documentation, bug reports, complaints, and even something you drew up on a napkin.
+Disable or enable metric logging for this specific plugin instance. By default we record all the metrics we can, but you can disable metrics collection for a specific plugin.
 
-Programming is not a required skill. Whatever you've seen about open source and maintainers or community members  saying "send patches or die" - you will not see that here.
+### `id`
 
-It is more important to the community that you are able to contribute.
+* Value type is string
+* There is no default value for this setting.
 
-For more information about contributing, see the [CONTRIBUTING](https://github.com/elastic/logstash/blob/master/CONTRIBUTING.md) file.
+Add a unique ID to the plugin configuration. If no ID is specified, Logstash will generate one. It is strongly recommended to set this ID in your configuration. This is particularly useful when you have two or more plugins of the same type. For example, if you have 2 datadog_metrics outputs. Adding a named ID in this case will help in monitoring Logstash when using the monitoring APIs.
+
+```
```

Comment:
```suggestion
```ruby
```

Reply:
or is this 
```suggestion
```json
```

Reply:
Like above, this is not ruby and it is not json. I can add ruby to get some basic syntax highlight though.